lib/collection/src/operations/conversions.rs

```diff
@@ -326,10 +329,37 @@ impl TryFrom<api::grpc::qdrant::VectorParams> for VectorParams {
             })?,
             distance: from_grpc_dist(vector_params.distance)?,
             hnsw_config: vector_params.hnsw_config.map(Into::into),
+            quantization_config: vector_params
+                .quantization_config
+                .map(grpc_to_segment_quantization_config),
         })
     }
 }
 
+fn grpc_to_segment_quantization_config(
+    value: api::grpc::qdrant::QuantizationConfig,
+) -> QuantizationConfig {
+    match value
+        .quantization
+        .expect("QuantizationConfig should always have a value")
+    {
+        api::grpc::qdrant::quantization_config::Quantization::Scalar(config) => {
+            QuantizationConfig::Scalar(ScalarQuantization {
+                scalar: ScalarQuantizationConfig {
+                    r#type: match QuantizationType::from_i32(config.r#type) {
+                        Some(QuantizationType::Int8) => ScalarType::Int8,
+                        Some(QuantizationType::UnknownQuantization) | None => {
+                            panic!("cannot convert ordering: {}", config.r#type)
```

Comment:
@generall We talked about this in a call. It uses `from_i32` to convert from `i32` in gRPC structs. This should never panic but we must cover all branches.

Reply:
let's return `Err` instead of panic